pkg/antctl/raw/traceflow/command.go

```diff
@@ -137,23 +138,31 @@ func runE(cmd *cobra.Command, _ []string) error {
 		return nil
 	}
 
-	if err := wait.Poll(1*time.Second, 15*time.Second, func() (bool, error) {
-		tf, err := client.OpsV1alpha1().Traceflows().Get(context.TODO(), tf.Name, metav1.GetOptions{})
+	var res *v1alpha1.Traceflow
+	err = wait.Poll(1*time.Second, 15*time.Second, func() (bool, error) {
+		res, err = client.OpsV1alpha1().Traceflows().Get(context.TODO(), tf.Name, metav1.GetOptions{})
 		if err != nil {
 			return false, err
 		}
-		if tf.Status.Phase != v1alpha1.Succeeded {
+		if res.Status.Phase != v1alpha1.Succeeded && res.Status.Phase != v1alpha1.Failed {
 			return false, nil
 		}
-		if err := output(tf); err != nil {
-			return false, fmt.Errorf("error when outputing result: %w", err)
-		}
 		return true, nil
-	}); err != nil {
+	})
+	if err == wait.ErrWaitTimeout {
+		err = errors.New("timeout waiting for Traceflow done")
```

Comment:
It seems error "timeout waiting for Traceflow done" is only for L156 to return. On L162, its value is overwritten by `output()`. Does it really output "timeout waiting for Traceflow done" after outputting partial traceflow result?
I think we can update code like below:
```
        var timeoutErr error
	if err == wait.ErrWaitTimeout {
		timeoutErr = errors.New("timeout waiting for Traceflow done")
		// Still output the Traceflow results if any.
		if res == nil {
			return timeoutErr
		}
	} else if err != nil {
		return fmt.Errorf("error when retrieving Traceflow: %w", err)
	}

	if err := output(res); err != nil {
		return fmt.Errorf("error when outputing result: %w", err)
	}
        if timeoutErr != nil {
                return timeoutErr
        }
	return nil
```

Reply:
This is not a mistake. Plesae ignore.